src/Microsoft.ML.StandardLearners/Standard/LinearPredictor.cs

```diff
@@ -185,7 +185,8 @@ protected LinearPredictor(IHostEnvironment env, string name, ModelLoadContext ct
                 _weightsDenseLock = new object();
         }
 
-        protected override void SaveCore(ModelSaveContext ctx)
+        [BestFriend]
+        private protected override void SaveCore(ModelSaveContext ctx)
```

Comment:
>private [](start = 8, length = 7)

This is already a protected function on an internal class -- isn't that already hidden?

Reply:
Sorry I misread - it is a public class and you are overriding SaveCore from PredictorBase which is protected. However, I dont quite follow how protected isnt hidden to begin with -- is this covering cases where classes derive from LinearPredictor? Also could LinearPredictor be internal?

---
In reply to: [236912196](https://github.com/dotnet/machinelearning/pull/1739#discussion_r236912196) [](ancestors = 236912196)

Reply:
Hi @singlis , protected is absolutely not hidden. It won't show up in usage of instances, but will show up in docs and other contexts.